Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -322,6 +322,21 @@ TEST_F(SerializerTest, BadStringValue) {
       Status(FirestoreErrorCode::InvalidArgument, "ignored"), bytes);
 }
 
+TEST_F(SerializerTest, BadStringValue2) {
+  std::vector<uint8_t> bytes;
+  Status status =
+      serializer.EncodeFieldValue(FieldValue::StringValue("a"), &bytes);
+  EXPECT_TRUE(status.ok());
+
+  // Mutate that bytes: Claim that the string length is 5 instead of 1.
+  // (The first two bytes are used by the encoded tag.)
+  EXPECT_EQ(bytes[2], 1);
```

Comment:
If a test can't reasonably proceed it's usually better to `ASSERT_FOO` rather than `EXPECT_FOO` so that you don't get a chain of random failures after a precondition doesn't hold.

In other words, use `EXPECT` for independent assertions and `ASSERT` for dependent ones.

In this case I'd suggest using ASSERT for both.